### PR TITLE
Array mirror inverse

### PR DESCRIPTION
**Pull request by TEJASWI-TEJASWI**
Implementation of Array Mirror Inverse

The inverse of an array means if the array elements are swapped with their corresponding indices and the array is called mirror-inverse if it's inverse is equal to itself.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given an array arr[], the task is to find whether the array is mirror inverse. The inverse of an array means if the array elements are swapped with their corresponding indices and the array is called mirror-inverse if it’s inverse is equal to itself.
A better approach is to traverse the array and for all the indices, if arr[arr[index]] = index is satisfied then the given array is mirror inverse.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#661

### CODE REVIEW

**Comment by TEJASWI-TEJASWI** [comment]
Please review my PR @HarshCasper @Kajol-Kumari @plazzy99 @Abhijit2505 @atarax665 @iamrajiv  @siddharth25pandey

File: C-Plus-Plus/ds/Array_mirror_inverse.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// C++ implementation of the approach
+#include<bits/stdc++.h>
+using namespace std;
+
+// Function that returns true if
+// the array is mirror-inverse
+bool mirr_inv(int arr[], int n)
```
Comment (iamrajiv) [inline C-Plus-Plus/ds/Array_mirror_inverse.cpp]: @TEJASWI-TEJASWI Use proper variables name like `mirr_inv` can be updated to `mirror_inverse`.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@TEJASWI-TEJASWI Format the code according to this https://google.github.io/styleguide/cppguide.html. Move this program to `cp` section. Update the README. Also, Add time and space complexity.

File: C-Plus-Plus/ds/Array_mirror_inverse.cpp (hunk 1 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Array_mirror_inverse.cpp]: I have update the variable name .
So please review @iamrajiv

Comment (iamrajiv) [reply C-Plus-Plus/ds/Array_mirror_inverse.cpp]: @TEJASWI-TEJASWI I have requested more changes.

Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Array_mirror_inverse.cpp]: I have resolved the issue.
please have a look on PR-969 @iamrajiv

Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Array_mirror_inverse.cpp]: I have update the readme file please have a look on PR-970 @iamrajiv

**Comment by iamrajiv** [comment]
@TEJASWI-TEJASWI Do not make new PR for every commit and changes. You can make changes in the same PR. Please make the requested changes in this PR.

File: C-Plus-Plus/ds/Array_mirror_inverse.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+enter elements 2 of array
+2
+enter elements 3 of array
+1
+enter elements 4 of array
+0
+My output of my sample input 
+My Array is Mirror inverse
+*/
+}
```
Comment (Kajol-Kumari) [inline C-Plus-Plus/ds/Array_mirror_inverse.cpp]: please shift the sample input output below this line i.e after code section ends.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Please update the reamde. Also the title `mirror inverse` doesn't seem to fit as I was expecting `123321` or `12321`to be an array mirror inverse but here we are doing a check for indices and the array element to be equal. I am not sure if this is the expected behaviour for this particular issue.
cc: @iamrajiv @plazzy99

File: C-Plus-Plus/ds/Array_mirror_inverse.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+int main()
+{
+		int n;
+		cout<<"Enter size of array ";
+		cin>>n;
+		int arr[n];
+		int k;
+		for(int j=0;j<n;j++)
+                {
```
Comment (iamrajiv) [inline C-Plus-Plus/ds/Array_mirror_inverse.cpp:26]: Take array input like this
```cpp
for (int i = 0; i < n; ++i) {
	cin >> arr[i];
}
```

File: C-Plus-Plus/ds/Array_mirror_inverse.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+                     cout<<"enter elements "<< j <<" of array";
+                     cin>>k;
+                     arr[j]=k;
+                }
+		if (mirror_inverse(arr,n))
+			cout << "My Array is Mirror inverse";
```
Comment (iamrajiv) [inline C-Plus-Plus/ds/Array_mirror_inverse.cpp]: Just output `Yes` or `No`  if the array is mirror inverse or not.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@TEJASWI-TEJASWI Move this program to `cp` section and update the README.

**Comment by iamrajiv** [comment]
@TEJASWI-TEJASWI Please make the requested changes.

**Comment by atarax665** [comment]
@TEJASWI-TEJASWI Haven't heard from you in a while. Closing the PR due to inactivity from your side.

**Pull request closed without merging**